tests/spark/test_spark_connect_model_export.py

```diff
@@ -0,0 +1,159 @@
+import json
+import os
+
+import numpy as np
+import pandas as pd
+import pyspark
+import pytest
+from packaging.version import Version
+from pyspark.ml.connect.classification import (
+    LogisticRegression as LORV2,
+)
+from pyspark.ml.connect.feature import StandardScaler
+from pyspark.ml.connect.pipeline import Pipeline
+from pyspark.sql import SparkSession
+from pyspark.sql import functions as spark_f
+from pyspark.sql.types import LongType
+from sklearn import datasets
+
+import mlflow
+import mlflow.pyfunc.scoring_server as pyfunc_scoring_server
+from mlflow import pyfunc
+from mlflow.pyfunc import spark_udf
+
+from tests.helper_functions import (
+    pyfunc_serve_and_score_model,
+)
+from tests.pyfunc.test_spark import score_model_as_udf
+from tests.spark.test_spark_model_export import (
+    SparkModelWithData,
+)
+
+if Version(pyspark.__version__) < Version("3.5"):
+    pytest.skip("pyspark ML connect Model is available in pyspark >= 3.5")
+
+
+def _get_spark_connect_session():
+    return (
+        SparkSession.builder.remote("local[2]")
+        .config("spark.connect.copyFromLocalToFs.allowDestLocal", "true")
+        .getOrCreate()
+    )
+
+
+@pytest.fixture
+def model_path(tmp_path):
+    return os.path.join(tmp_path, "model")
+
+
+def score_model_as_udf(model_uri, pandas_df, result_type):
+    spark = SparkSession.getActiveSession()
+    spark_df = spark.createDataFrame(pandas_df).coalesce(1)
+    pyfunc_udf = spark_udf(
+        spark=spark, model_uri=model_uri, result_type=result_type, env_manager="local"
+    )
+    new_df = spark_df.withColumn("prediction", pyfunc_udf(spark_f.struct(spark_f.col("features"))))
+    return new_df.toPandas()["prediction"]
+
+
+@pytest.fixture(scope="module")
+def spark():
+    spark = _get_spark_connect_session()
+    yield spark
+    spark.stop()
+
+
+@pytest.fixture(scope="module")
+def iris_df(spark):
+    X, y = datasets.load_iris(return_X_y=True)
+    spark_df = spark.createDataFrame(list(zip(X, y)), schema="features: array<double>, label: long")
+    return spark_df.toPandas(), spark_df
+
+
+@pytest.fixture(scope="module")
+def spark_model_iris(iris_df):
+    iris_pandas_df, iris_spark_df = iris_df
+    scaler = StandardScaler(inputCol="features", outputCol="scaled_features")
+    lr = LORV2(maxIter=50, numTrainWorkers=2, learningRate=0.001)
+    pipeline = Pipeline(stages=[scaler, lr])
+    # Fit the model
+    model = pipeline.fit(iris_spark_df)
+    preds_pandas_df = model.transform(iris_pandas_df.copy(deep=False))
+    return SparkModelWithData(
+        model=model,
+        spark_df=iris_spark_df,
```

Comment:
Looks like we're not using this field (`spark_df`) at all.

Reply:
Yes. But not a big deal, we might use it in future, and we can reuse SparkModelWithData namedtuple

Reply:
> we might use it in future

https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it